packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md

```diff
@@ -0,0 +1,154 @@
+        } else {
+          VStack {
+            Button("Sign Out") {
+              signOut()
+            }
+            .padding(.bottom)
+            HStack {
+              Button("Show User Info") {
+                showUserInfo()
+              }
+              Button("Show Token") {
+                showTokenInfo()
+              }
+              .padding(.horizontal)
+              Button("Refresh Token") {
+                refreshToken()
+              }
+            }
+          }
+        }
+      }
+      Divider()
+      Text("Status")
+      .fontWeight(.medium)
+      .foregroundColor(Color.gray)
+      .padding(.bottom)
+      Text(statusText ?? "Unknown status.")
+      Divider()
+      Text("Info")
+      .fontWeight(.medium)
+      .foregroundColor(Color.gray)
+      .padding(.bottom)
+      ScrollView(showsIndicators: true) {
+        Text(infoText ?? "")
+      }
+    }
+    .alert(isPresented: $showingError) {
+      Alert(title: Text(errorTitle ?? "Unknown Error"),
+      message: Text(errorMessage ?? "An unknow error occured."))
+    }
+    .onAppear() {
+      configureSDK()
+    }
+  }
+
+  func signIn() {
+    statusText = "Sign in button touched."
+    signedIn = true
+  }
+
+  func signOut() {
+    statusText = "Sign out button touched."
+    signedIn = false
+  }
+
+  func showUserInfo() {
+    statusText = "Show user info button touched"
+  }
+
+  func showTokenInfo() {
+    statusText = "Show token info button touched"
+  }
+
+  func refreshToken() {
+    statusText = "Refresh token button touched"
+  }
+
+  func configureSDK() {
+    print("configureSDK called")
+  }
+
+  func tokenExpired() {
+    print("tokenExpired called")
+  }
+
+  func showError(title: String , message: String) {
+    errorTitle = title
+    errorMessage = message
+    showingError = true
+  }
+
+  func showError(title: String, error: Error) {
+    showError(title: title, 
+    message: error.localizedDescription)
+  }
+
+  func updateStatus(_ statusText: String, infoText: String, signedInStatus: Bool) {
+    self.statusText = statusText
+    self.infoText = infoText
+    signedIn = signedInStatus
+  }
+
+}
+```
+
+The last four functions are utilities. You fill out `tokenExpired` later. The two variants of `showError` configure the contents, and then show an alert (the `showingError` state variable controls the presentation of the alert.)
+
+The last utility updates the status text, the info text, and the `signedIn` state variable that controls the buttons presented at the top of the UI.
+
+#### Add an extension to find the root view
+
+Redirect sign-in is designed for apps using view controllers and storyboards. Some of the calls include a parameter for the current view controller, something that doesn't exist in SwiftUI. This sample uses the root view controller for the first key window of the active app for the parameter.
+
+Add a new Swift file to your project called `UIApplication+rootViewController` and add the following code:
```

Comment:
I think that sentence needs a lead in:

To find the root view controller, add a new...

Reply:
updated